src/components/ConfigEditor/AzureCredentialsConfig.ts

```diff
@@ -0,0 +1,197 @@
+import { DataSourceSettings } from '@grafana/data';
+import { config } from '@grafana/runtime';
+
+import { AzureCloud, AzureCredentials, ConcealedSecret } from './AzureCredentials';
+
+const concealed: ConcealedSecret = Symbol('Concealed client secret');
+
+export const getOboEnabled = (): boolean => !!config.featureToggles['adxOnBehalfOf'];
+
+function getDefaultAzureCloud(): string {
+  return config.azure.cloud || AzureCloud.Public;
+}
+
+function getSecret(options: DataSourceSettings<any, any>): undefined | string | ConcealedSecret {
+  if (options.secureJsonFields.azureClientSecret) {
+    // The secret is concealed on server
+    return concealed;
+  } else {
+    const secret = options.secureJsonData?.azureClientSecret;
+    return typeof secret === 'string' && secret.length > 0 ? secret : undefined;
+  }
+}
+
+export function getDefaultCredentials(): AzureCredentials {
+  if (config.azure.managedIdentityEnabled) {
+    return { authType: 'msi' };
+  } else {
+    return { authType: 'clientsecret', azureCloud: getDefaultAzureCloud() };
+  }
+}
+
+export function getCredentials(options: DataSourceSettings<any, any>): AzureCredentials {
+  const credentials = options.jsonData.azureCredentials as AzureCredentials | undefined;
+
+  // If no credentials saved then try to restore legacy credentials, otherwise return default credentials
+  if (!credentials) {
+    return getLegacyCredentials(options) ?? getDefaultCredentials();
+  }
+
+  switch (credentials.authType) {
+    case 'msi':
+      if (config.azure.managedIdentityEnabled) {
+        return {
+          authType: 'msi',
+        };
+      } else {
+        // If authentication type is managed identity but managed identities were disabled in Grafana config,
+        // then we should fall back to an empty default credentials
+        return getDefaultCredentials();
+      }
+    case 'clientsecret':
+    case 'clientsecret-obo':
+      if (credentials.authType === 'clientsecret-obo' && !getOboEnabled()) {
+        // If authentication type is OBO but OBO were disabled in Grafana config,
+        // then we should fall back to an empty default credentials
+        return getDefaultCredentials();
+      }
```

Comment:
My IDE (ESLint?) doesn't like it. And I agree, it's too dangerous, it can be accidentally changed in future, someone may rearrange the order of cases without noticing that this is the fall thru, and everything will get broken. Let's keep it more straightforward. even if it's checking same thing two times.